contracts/parent-chain/espresso-migration/DeployAndInitEspressoSequencerInbox.s.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "forge-std/Script.sol";
+import "nitro-contracts/bridge/SequencerInbox.sol";
+import "nitro-contracts/bridge/ISequencerInbox.sol";
+
+/// @notice This contract deploys and initializes a sequencerInbox contract that orbit chains can migrate to that enables compatibility
+/// with the espresso confirmation layer
+/// @dev BATCH_POSTER_ADDRS should be a comma delimited list that includes addresses. This list will give batch posting affordances to those addresses
+///        For chains using the Espresso TEE integration, this will be the address of your new batch poster, if you decide to change it.
+contract DeployAndInitEspressoSequencerInbox is Script {
+    function run() external {
+        // Grab addresses from env
+        address bridgeAddr = vm.envAddress("BRIDGE_ADDRESS");
+        address reader4844Addr = vm.envAddress("READER_ADDRESS");
+        address espressoTeeVerifierAddr = vm.envAddress("ESPRESSO_TEE_VERIFIER_ADDRESS");
+
+        // Grab the list of batch poster addresses valid for this rollup. This is a comma delimited list of addresses.
+        address[] memory batchPosters = vm.envAddress("BATCH_POSTER_ADDRS", ",");
+        address batchPosterManager = vm.envAddress("BATCH_POSTER_MANAGER");
+
+        // Grab any uints we need to initialize the contract from envAddress
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+        uint256 delayBlocks = vm.envUint("DELAY_BLOCKS");
+        uint256 futureBlocks = vm.envUint("FUTURE_BLOCKS");
+        uint256 delaySeconds = vm.envUint("DELAY_SECONDS");
+        uint256 futureSeconds = vm.envUint("FUTURE_SECONDS");
+        uint256 maxDataSize = vm.envUint("MAX_DATA_SIZE");
+        // Grab booleans we need from env
+        bool isUsingFeeToken = vm.envBool("IS_USING_FEE_TOKEN");
+
+        // initialize interfaces needed
+        IReader4844 reader = IReader4844(reader4844Addr);
+        IBridge bridge = IBridge(bridgeAddr);
+        // Start broadcast to deploy and initializer the sequencerInbox
+        vm.startBroadcast(deployerPrivateKey);
+        SequencerInbox sequencerInbox = new SequencerInbox(maxDataSize, reader, isUsingFeeToken);
+        ISequencerInbox.MaxTimeVariation memory maxTimeVariation = ISequencerInbox.MaxTimeVariation({
+            delayBlocks: delayBlocks,
+            futureBlocks: futureBlocks,
+            delaySeconds: delaySeconds,
+            futureSeconds: futureSeconds
+        });
+        sequencerInbox.initialize(bridge, maxTimeVariation, espressoTeeVerifierAddr);
```

Comment:
you cannot call initialize on implementation contract.

Reply:
This is because of the onlyDelegated requirement correct?

Reply:
you should never call implementation contract's functions directly. It should always be through proxy. 

Can you read this? https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable Its going to give you a better understanding of how they work.